apps/api/lib/api/lottery.ex

```diff
@@ -0,0 +1,124 @@
+    total_distance = total_distance(players)
+    number_of_players = length(players)
+    Enum.reduce(players, [], fn(player, acc) ->
+      number_of_tickets = qualifying_calculation(player, total_distance, number_of_players)
+      tickets = List.duplicate(player.id, number_of_tickets)
+      [tickets | acc]
+    end)
+  end
+
+  # Builds the probability pool for players in a non-qualifying round.
+  #
+  # It finds the average distance of the zombie team, calculates a probability
+  # for each player based on their distance relative to the zombie average, and
+  # adds "lottery tickets" to the pool.
+  defp build_round_pool(players) do
+    zombie_avg = average_distance(players, is_human: false)
+    humans = filter_players(players)
+    Enum.reduce(humans, [], fn(player, acc) ->
+      tickets = determine_player_tickets(player, zombie_avg)
+      [tickets | acc]
+    end)
+  end
+
+  # Calculates a player's number of tickets for the qualifying round.
+  #
+  # A player's distance is weighed against the total distance of all players
+  # in the round.
+  defp qualifying_calculation(player, total_distance, number_of_players) do
+    round(((1 - (player.stats.distance / total_distance)) * number_of_players))
+  end
+
+  # Determines the number of tickets that a player has in the lottery.
+  #
+  # The methodology is as follows: A player starts with 10 tickets, and
+  # is given 'safe tickets' for distance run beyond the calculated average
+  # distance of the zombie team. If a player runs >= twice the amount of the
+  # zombie average, they are completely safe: All 10 of their tickets are
+  # added as 'safe tickets,' without the player's id on them. If the player does
+  # not pass the zombie average, all tickets are returned with the player's id.
+  defp determine_player_tickets(player, zombie_average) do
+    starting = 10
+    safe = calculate_safe_tickets(player.stats.distance,
+                                  zombie_average,
+                                  starting)
+    safe_tickets = List.duplicate(:safe, safe)
+    player_tickets = List.duplicate(player.id, starting - safe)
+    Enum.concat(safe_tickets, player_tickets)
+  end
+
+  # Determines the number of "safe tickets" available to a player in a
+  # non-qualifying round.
+  #
+  # Iff a player's distance is greater than the zombie average, they will
+  # be awarded "safe tickets."
+  defp calculate_safe_tickets(distance, zombie_average, starting)
+    when distance > zombie_average do
+      min(round(starting * ((distance / zombie_average) - 1)), starting)
+  end
+  defp calculate_safe_tickets(_distance, _zombie_average, _starting) do
+    0
+  end
+
+  # Determines the total distance of a given List of Players.
+  defp total_distance(players) do
+    Enum.reduce(players, 0, fn(player, acc) ->
+      acc + player.stats.distance
+    end)
+  end
+
+  # Filters players by their `is_human` status.
+  defp filter_players(players, [{:is_human, is_human}]) do
```

Comment:
we could consolidate these functions and default the is_human param. the two function heads might be more performant though